azurelinuxagent/common/protocol/wire.py

```diff
@@ -649,6 +657,9 @@ def fetch_manifest(self, version_uris):
                 else:
                     host.manifest_uri = version.uri
                     logger.verbose("Manifest downloaded successfully from host plugin")
+                    if not HostPluginProtocol.is_default_channel():
+                        logger.info("Setting host plugin as default channel")
+                        HostPluginProtocol.set_default_channel(True)
```

Comment:
You may want to push this into the HostPluginProtocol: It should become the default channel any time it is used and succeeds. Once default, it stays default, even if subsequent errors occur.